fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryIndexPlan.java

```diff
@@ -159,24 +153,28 @@ private RecordQueryIndexPlan(@Nonnull final String indexName,
     @Override
     public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store, @Nonnull final EvaluationContext context,
                                                                      @Nullable final byte[] continuation, @Nonnull final ExecuteProperties executeProperties) {
-        switch (useIndexPrefetch) {
-            case NONE:
+        RecordQueryPlannerConfiguration.IndexFetchMethod fetchMethod = indexFetchMethod;
+        // Check here to allow for the store API_VERSION to change
+        if (!store.getContext().getAPIVersion().isAtLeast(APIVersion.API_VERSION_7_1)) {
+            logWarning("Index remote fetch can only be used with API_VERSION of at least 7.1. Falling back to regular scan.");
```

Comment:
This feels like it should only warn if the `fetchMethod` is `USE_REMOTE_FETCH` or maybe `USE_REMOTE_FETCH_AND_FALLBACK`. But it definitely shouldn't warn if the `fetchMethod` is `SCAN_AND_FETCH`